docs/rfcs/826-add-multipart-http-signer-support.md

```diff
@@ -0,0 +1,70 @@
+- Author: abyss-w <mad.hatter@foxmail.com>
+- Start Date: 2021-09-26
+- RFC PR: [beyondstorage/go-storage#826](https://github.com/beyondstorage/go-storage/pull/826)
+- Tracking Issue: [beyondstorage/go-storage#0](https://github.com/beyondstorage/go-storage/issues/0)
+
+# GSP-826: Add Multipart HTTP Signer Support
+
+Previous discussion:
+
+- [Multipart related operations ](https://forum.beyondstorage.io/t/topic/226)
+
+# Background
+
+In [GSP-729](https://github.com/beyondstorage/go-storage/blob/master/docs/rfcs/729-redesign-http-signer.md), we split out the `HTTPSigner` proposed in [GSP-706](https://github.com/beyondstorage/go-storage/blob/master/docs/rfcs/706-support-http-signer.md). We had the following problems in the implementation of GSP-706:
+
+- There's no appropriate way to pass in some parameters for some operations like [How to pass partIndex into QuerySignHTTP for WriteMultipart](https://forum.beyondstorage.io/t/how-to-pass-partindex-into-querysignhttp-for-writemultipart/192).
+- `CreateMultipart` needs to support Expires parameter.
+
+## Proposal
+
+I propose to add `MultipartHTTPSigner` interface.
+
+```go
+type MultipartHTTPSigner interface {
+signHTTPCreateMultipart(path string, expire time.Duration, ps ...types.Paires) (req *http.Request, err error)
+signHTTPWriteMultipart(o *Object, size int64, index int, expire time.Duration, ps ...types.Pairs) (req *http.Request, err error)
+}
```

Comment:
The added interfaces should be public. How about `QuerySignHTTPCreateMultipart` and `QuerySignHTTPWriteMultipart`? And the format(indent) should be updated.